consensus/hotstuff/voteaggregator/vote_aggregator_v2.go

```diff
@@ -63,48 +65,43 @@ func NewVoteAggregatorV2(
 		queuedVotesNotifier: engine.NewNotifier(),
 	}
 
-	return aggregator, nil
-}
+	componentBuilder := component.NewComponentManagerBuilder()
 
-// Ready returns a ready channel that is closed once the engine has fully
-// started. For the propagation engine, we consider the engine up and running
-// upon initialization.
-func (va *VoteAggregatorV2) Ready() <-chan struct{} {
-	va.lm.OnStart(func() {
-		// launch as many workers as we need
-		for i := 0; i < defaultVoteAggregatorWorkers; i++ {
-			va.unit.Launch(va.queuedVotesProcessingLoop)
-		}
-	})
-	return va.lm.Started()
-}
+	// launch as many workers as we need
+	for i := 0; i < defaultVoteAggregatorWorkers; i++ {
+		componentBuilder.AddWorker(func(ctx irrecoverable.SignalerContext, ready component.ReadyFunc) {
+			ready()
+			err := aggregator.queuedVotesProcessingLoop(ctx)
+			if err != nil {
+				ctx.Throw(err)
+			}
+		})
+	}
```

Comment:
I think we could merge the abstract function `func(ctx irrecoverable.SignalerContext, ready component.ReadyFunc) { ... }` into `queuedVotesProcessingLoop`, which essentially is the worker function. Why not make it conform to [`ComponentWorker`](https://github.com/onflow/flow-go/blob/50f0ba0e46898c6dacab9738d2fd946b07f3c99d/module/component/component.go#L152) (?)

```suggestion
	// initialize ComponentManager with workers to process votes
	componentBuilder := component.NewComponentManagerBuilder()
	for i := 0; i < defaultVoteAggregatorWorkers; i++ {
		componentBuilder.AddWorker(aggregator.queuedVotesProcessingLoop)
	}
	aggregator.ComponentManager = componentBuilder.Build()
```

where the method `queuedVotesProcessingLoop` could look something like
```golang
// queuedVotesProcessingLoop implements `component.ComponentWorker`.
// The worker will signal its readiness, and commence processing votes
// immediately when `ready` returns. The worker terminates when ctx.Done() returns
// or when it encounters an unexpected exception while processing votes.
func (va *VoteAggregatorV2) queuedVotesProcessingLoop(ctx irrecoverable.SignalerContext, ready component.ReadyFunc) {
	notifier := va.queuedVotesNotifier.Channel()
	ready() // signal that this worker is ready
	for {
		select {
		case <-ctx.Done():
			return
		case <-notifier:
			err := va.processQueuedVoteEvents(ctx)
			if err != nil {
				ctx.Throw(
					fmt.Errorf("internal error processing queued vote events: %w", err),
				)
				return
			}
		}
	}
}
```

Reply:
My main objection would be to make this function to return error instead of handle it, but it makes initialization code cleaner so let's give it a try.